Sources/BaggageContext/Context.swift

```diff
@@ -0,0 +1,287 @@
+    ///
+    /// - Parameter logger: Logger that should replace the underlying logger of this context.
+    /// - Returns: new context, with the passed in `logger`
+    public func withLogger(_ logger: Logger) -> Context {
+        var copy = self
+        copy.logger = logger
+        return copy
+    }
+
+    /// Fluent API allowing for modification of underlying log level when passing the context to other functions.
+    ///
+    /// - Parameter logLevel: New log level which should be used to create the new context
+    /// - Returns: new context, with the passed in `logLevel` used for the underlying logger
+    public func withLogLevel(_ logLevel: Logger.Level) -> Context {
+        var copy = self
+        copy.logger.logLevel = logLevel
+        return copy
+    }
+
+    /// Fluent API allowing for modification a few baggage values when passing the context to other functions, e.g.
+    ///
+    ///     makeRequest(url, context: context.withBaggage {
+    ///         $0.traceID = "fake-value"
+    ///         $0.calledFrom = #function
+    ///     })
+    ///
+    /// - Parameter function:
+    public func withBaggage(_ function: (inout Baggage) -> Void) -> Context {
+        var baggage = self.baggage
+        function(&baggage)
+        return self.withBaggage(baggage)
+    }
+
+    /// Fluent API allowing for replacement of underlying baggage when passing the context to other functions.
+    ///
+    /// - Warning: Use with caution, generally it is not recommended to modify an entire baggage, but rather only add a few values to it.
+    ///
+    /// - Parameter baggage: baggage that should *replace* the context's current baggage.
+    /// - Returns: new context, with the passed in baggage
+    public func withBaggage(_ baggage: Baggage) -> Context {
+        var copy = self
+        copy.baggage = baggage
+        return copy
+    }
+}
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: Context Initializers
+
+extension Context {
+    /// An empty baggage context intended as the "root" or "initial" baggage context background processing tasks, or as the "root" baggage context.
+    ///
+    /// It is never canceled, has no values, and has no deadline.
+    /// It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.
+    ///
+    /// ### Usage in frameworks and libraries
+    /// This function is really only intended to be used frameworks and libraries, at the "top-level" where a request's,
+    /// message's or task's processing is initiated. For example, a framework handling requests, should create an empty
+    /// context when handling a request only to immediately populate it with useful trace information extracted from e.g.
+    /// request headers.
+    ///
+    /// ### Usage in applications
```

Comment:
may be required in testing?

Reply:
Right yes, I call this out a few lines above as:

>     /// It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

Should I rephrase the "Usage in applications" or add an usage section for testing that it's ok to create one there too?

Reply:
I guess its fine